airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_29_0.java

```diff
@@ -0,0 +1,135 @@
+  private static final ResourceId OPERATION_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC_OPERATION");
+  private static final ImmutableSet<ResourceId> RESOURCE_WITH_WORKSPACE_ID = ImmutableSet.of(
+      WORKSPACE_RESOURCE_ID,
+      SOURCE_RESOURCE_ID,
+      DESTINATION_RESOURCE_ID,
+      OPERATION_RESOURCE_ID);
+
+  private static final ResourceId CONNECTION_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC");
+  private static final Set<String> NAMESPACE_DEFINITIONS = ImmutableSet.of("source", "destination", "customformat");
+
+  private final Migration previousMigration;
+  private final Supplier<UUID> uuidSupplier;
+
+  public MigrationV0_29_0(Migration previousMigration) {
+    this(previousMigration, UUID::randomUUID);
+  }
+
+  @VisibleForTesting
+  public MigrationV0_29_0(Migration previousMigration, Supplier<UUID> uuidSupplier) {
+    super(previousMigration);
+    this.previousMigration = previousMigration;
+    this.uuidSupplier = uuidSupplier;
+  }
+
+  @Override
+  public String getVersion() {
+    return MIGRATION_VERSION;
+  }
+
+  @Override
+  public Map<ResourceId, JsonNode> getOutputSchema() {
+    final Map<ResourceId, JsonNode> outputSchema = new HashMap<>(previousMigration.getOutputSchema());
+    outputSchema.put(CONNECTION_RESOURCE_ID, MigrationUtils.getSchemaFromResourcePath(RESOURCE_PATH, CONNECTION_RESOURCE_ID));
+    return outputSchema;
+  }
+
+  @Override
+  public void migrate(Map<ResourceId, Stream<JsonNode>> inputData,
+                      Map<ResourceId, Consumer<JsonNode>> outputData) {
+    final UUID newWorkspaceId = uuidSupplier.get();
+
+    for (final Map.Entry<ResourceId, Stream<JsonNode>> entry : inputData.entrySet()) {
+      final Consumer<JsonNode> recordConsumer = outputData.get(entry.getKey());
+      entry.getValue().forEach(r -> {
+        // if standard sync make sure namespaceDefinition is set. it is now a required property. default to
+        // destination.
+        if (entry.getKey().equals(CONNECTION_RESOURCE_ID)) {
+          if (!r.hasNonNull("namespaceDefinition") || !NAMESPACE_DEFINITIONS.contains(r.get("namespaceDefinition").asText())) {
+            LOGGER.info(
+                "For connection {}, found namespaceDefinition {}. setting it to \"destination\"",
+                r.get("connectionId"),
+                r.get("namespaceDefinition"));
+            ((ObjectNode) r).put("namespaceDefinition", "destination");
+          }
+        }
+
+        // if the resource contains a workspace id. figure out if it is the default one and switch it.
+        if (RESOURCE_WITH_WORKSPACE_ID.contains(entry.getKey())) {
+          final UUID workspaceId = UUID.fromString(r.get("workspaceId").asText());
+          if (workspaceId.equals(DEFAULT_WORKSPACE_ID)) {
+            ((ObjectNode) r).put("workspaceId", newWorkspaceId.toString());
+          }
+        }
```

Comment:
The new workspace ID confused one user. A suggestion for the future: we should explicitly log the old and new workspace IDs so it is easier to debug.